ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -191,6 +194,24 @@ func (p *LocalIgnitionProvider) GetPayload(ctx context.Context, releaseImage str
 		}
 	}
 
+	isMultiArchManifestList, err := registryclient.IsMultiArchManifestList(ctx, mcoImage, pullSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine if image is manifest listed: %w", err)
```

Comment:
I'm surprise there's no generic library which does this for us?

Reply:
I've looked around in the `oc` repo and not seen anything really solid. I have seen [code](https://github.com/openshift/oc/blob/27ece46d94586eb4acb147cf96992f75c071c64e/pkg/oc/cli/image/manifest/manifest.go#L206) that looks at digest length which suggests one could maybe look at that.

Reply:
Could be mistaken, but it looks like they did something [similar](https://github.com/dmcgowan/containerd/blob/9edcfcc1cb4774d34768c6fcf1f1e30595e9af9d/images/handlers.go#L200) to that oc reference from above